iceoryx_platform/generic/include/iceoryx_platform/atomic.hpp

```diff
@@ -0,0 +1,518 @@
+// Copyright (c) 2024 by ekxide IO GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_PLATFORM_ATOMIC_HPP
+#define IOX_PLATFORM_ATOMIC_HPP
+
+#include <atomic>
+#include <type_traits>
+
+namespace iox::concurrent
+{
+/// @brief An alias to the std::atomic_flag
+using AtomicFlag = std::atomic_flag;
+
+/// @brief A thin wrapper for a 'std::atomic' which ensures that all atomic operations are always lock-free in order to
+/// ensure safe usage across processes in shared memory. For detailed documentation please have a look at
+/// https://en.cppreference.com/w/cpp/atomic/atomic
+template <typename T>
+class Atomic
+{
+  private:
+    template <typename U>
+    using enable_if_integral_t = std::enable_if_t<std::is_same_v<T, U> && std::is_integral_v<U>, T>;
+
+    template <typename U>
+    using enable_if_pointer_t = std::enable_if_t<std::is_same_v<T, U> && std::is_pointer_v<U>, T>;
+
+    template <typename U>
+    using enable_if_integral_or_pointer_t =
+        std::enable_if_t<std::is_same_v<T, U> && (std::is_integral_v<U> || std::is_pointer_v<U>), T>;
+
+  public:
+    static_assert(std::atomic<T>::is_always_lock_free,
+                  "The 'iox::Atomic' must work across process boundaries and must therefore be always lock-free!");
+
+    /// @brief This is always 'true' for 'iox::Atomic'
+    static constexpr bool is_always_lock_free = std::atomic<T>::is_always_lock_free;
+
+    /// @brief Constructs a new 'iox::Atomic' with a default value
+    constexpr Atomic() noexcept
+        : m_value{T()}
+    {
+    }
+
+    /// @brief Constructs a new 'iox::Atomic' with the given value
+    explicit constexpr Atomic(T value) noexcept
```

Comment:
This is a deviation from std::atomic but clang-tidy complains when it is not explicit